src/sys.c

```diff
@@ -587,6 +587,40 @@ typedef DWORD (WINAPI *GAPC)(WORD);
 #endif
 #endif
 
+int num_threads_bound(void) JL_NOTSAFEPOINT
+{
+#ifdef _OS_DARWIN_
+    return INT_MAX;
+#else
+    int masksize = uv_cpumask_size();
+    uv_thread_t tid = uv_thread_self();
+    char *cpumask = (char *)calloc(masksize, sizeof(char));
+    int err = uv_thread_getaffinity(&tid, cpumask, masksize);
+    if (err) {
+        free(cpumask);
+        jl_safe_printf("WARNING: failed to get thread affinity (%s %d)\n", uv_err_name(err),
+                       err);
+        return INT_MAX;
+    }
+    int n = 0;
+    for (size_t i = 0; i < masksize; i++) {
+        n += cpumask[i];
+    }
+    free(cpumask);
+    return n;
+#endif
+}
+
+int intmin(int a, int b) JL_NOTSAFEPOINT
```

Comment:
Can't you just use a ternary expression, like `upper_bound < count ? upper_bound : count`?

Reply:
Yeah, maybe that's better. I didn't want to repeat things, but that expression is not really long.